presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetColumnReader.java

```diff
@@ -170,32 +170,41 @@ public Block readPrimitive(Type type, IntList positions)
     }
 
     private void readValues(BlockBuilder blockBuilder, int numValues, Type type, IntList positions)
+            throws IOException
     {
-        definitionLevel = definitionReader.readLevel();
-        repetitionLevel = repetitionReader.readLevel();
-        int valueCount = 0;
-        for (int i = 0; i < numValues; i++) {
-            do {
-                readValue(blockBuilder, type);
-                try {
-                    valueCount++;
-                    repetitionLevel = repetitionReader.readLevel();
-                    if (repetitionLevel == 0) {
-                        positions.add(valueCount);
-                        valueCount = 0;
-                        if (i == numValues - 1) {
-                            return;
-                        }
+        int rowElementsCount = 0;
```

Comment:
Please rename these variables to be more descriptive. Currently it's hard to follow what they do. Also in this part of the code we use many different terms like element, position, value, row. Please also make your use of these words consistent.

IIUC, `rowsCount` refers to the total number of values read so far, and `totalCount` refers to the number of values read in the current page page (something like `valuesReadInCurrentPage`). I am not sure about `rowElementsCount`, what does it mean for primitives?